linchpin/shell/__init__.py

```diff
@@ -282,11 +295,15 @@ def destroy(ctx, targets, run_id, tx_id):
     the appropriate PinFile will be destroyed.
 
     """
+    flags = {}
+    flags["run_hooks_on_failure"] = run_hooks_on_failure
+    flags["no_hooks"] = no_hooks
```

Comment:
See the example in the up() method for how to handle this without passing more things around.